File: schemas/recipe-cooking/drizzle/recipe_nutrition.ts
// recipe_nutrition: Per-recipe macronutrient and micronutrient summary values.
// See README.md for full design rationale.

import { pgTable, uuid, numeric, timestamp } from "drizzle-orm/pg-core";
import { recipes } from "./recipes";

export const recipeNutrition = pgTable(
  "recipe_nutrition",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    recipeId: uuid("recipe_id").notNull().unique().references(() => recipes.id, { onDelete: "cascade" }),
    calories: numeric("calories"),
    totalFatGrams: numeric("total_fat_grams"),
    saturatedFatGrams: numeric("saturated_fat_grams"),
    carbohydratesGrams: numeric("carbohydrates_grams"),
    fiberGrams: numeric("fiber_grams"),
    sugarGrams: numeric("sugar_grams"),
    proteinGrams: numeric("protein_grams"),
    sodiumMg: numeric("sodium_mg"),
    cholesterolMg: numeric("cholesterol_mg"),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  }
);

File: schemas/recipe-cooking/spacetimedb/recipe_nutrition.rs
// recipe_nutrition: Nutritional information per recipe serving.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = recipe_nutrition, public)]
pub struct RecipeNutrition {
    #[primary_key]
    pub id: String, // UUID

    #[unique]
    pub recipe_id: String, // FK → recipes.id (cascade delete)

    pub calories: Option<f64>,
    pub total_fat_grams: Option<f64>,
    pub saturated_fat_grams: Option<f64>,
    pub carbohydrates_grams: Option<f64>,
    pub fiber_grams: Option<f64>,
    pub sugar_grams: Option<f64>,
    pub protein_grams: Option<f64>,
    pub sodium_mg: Option<f64>,
    pub cholesterol_mg: Option<f64>,
    pub updated_at: Timestamp,
}
